tests/TopDownProteomics.Tests/ProFormaParserTests.cs

```diff
@@ -193,77 +219,143 @@ public void Rule6_7_Invalid(string proFormaString)
+            const string proFormaString = "[Unimod]+[1]-S[21]GRGK[1]QGGKARAKAKTRSSRAGKVTIAQGGVLPNIQAVLLPKKT[21]ESHHKAKGK";
+            var term = _parser.ParseString(proFormaString);
+
+            Assert.AreEqual("SGRGKQGGKARAKAKTRSSRAGKVTIAQGGVLPNIQAVLLPKKTESHHKAKGK", term.Sequence);
+            Assert.IsNotNull(term.Tags);
+            Assert.AreEqual(3, term.Tags.Count);
+            Assert.IsNotNull(term.NTerminalDescriptors);
+            Assert.AreEqual(1, term.NTerminalDescriptors.Count);
+            Assert.IsNull(term.CTerminalDescriptors);
+
+            var nTerm = term.NTerminalDescriptors[0];
+            Assert.AreEqual(ProFormaKey.Unimod, nTerm.Key);
+            Assert.AreEqual("1", nTerm.Value);
+
+            ProFormaTag tag1 = term.Tags[0];
+            Assert.AreEqual(0, tag1.Index);
+            Assert.AreEqual(1, tag1.Descriptors.Count);
+            Assert.AreEqual(ProFormaKey.Unimod, tag1.Descriptors.Single().Key);
+            Assert.AreEqual("21", tag1.Descriptors.Single().Value);
+
+            ProFormaTag tag5 = term.Tags[1];
+            Assert.AreEqual(4, tag5.Index);
+            Assert.AreEqual(1, tag5.Descriptors.Count);
+            Assert.AreEqual(ProFormaKey.Unimod, tag5.Descriptors.Single().Key);
+            Assert.AreEqual("1", tag5.Descriptors.Single().Value);
+
+            ProFormaTag tag44 = term.Tags[2];
+            Assert.AreEqual(43, tag44.Index);
+            Assert.AreEqual(1, tag44.Descriptors.Count);
+            Assert.AreEqual(ProFormaKey.Unimod, tag44.Descriptors.Single().Key);
+            Assert.AreEqual("21", tag44.Descriptors.Single().Value);
         }
 
         [Test]
-        public void EmptyDescriptor()
+        public void BestPractice_iii()
         {
-            const string proFormaString = "PRO[]TEOFORM";
-            Assert.Throws<ProFormaParseException>(() => _parser.ParseString(proFormaString));
+            const string proFormaString = "MTLFQLREHWFVYKDDEKLTAFRNK[p-adenosine| N6-(phospho-5'-adenosine)-L-lysine(RESID)| RESID:AA0227| PSI-MOD:MOD:00232| N6AMPLys(PSI-MOD)]SMLFQRELRPNEEVTWK";
+            var term = _parser.ParseString(proFormaString);
+
+            Assert.AreEqual("MTLFQLREHWFVYKDDEKLTAFRNKSMLFQRELRPNEEVTWK", term.Sequence);
+            Assert.IsNotNull(term.Tags);
+            Assert.AreEqual(1, term.Tags.Count);
+            Assert.IsNull(term.NTerminalDescriptors);
+            Assert.IsNull(term.CTerminalDescriptors);
+
+            ProFormaTag tag25 = term.Tags[0];
+            Assert.AreEqual(24, tag25.Index);
+            Assert.AreEqual(5, tag25.Descriptors.Count);
+            Assert.AreEqual(ProFormaKey.Mod, tag25.Descriptors[0].Key);
+            Assert.AreEqual("p-adenosine", tag25.Descriptors[0].Value);
+            Assert.AreEqual(ProFormaKey.Mod, tag25.Descriptors[1].Key);
+            Assert.AreEqual(" N6-(phospho-5'-adenosine)-L-lysine(RESID)", tag25.Descriptors[1].Value);
```

Comment:
Should this have the space trimmed off of the left?

Reply:
I think this should be trimmed, so that using string comparisons with modifications in a database would give expected results.

Reply:
Open question in my opinion. Right now I'm only trimming from the ends of the descriptor. I am preserving all of the spaces directly after the colon. Seems to be the middle ground and I'm not sure if that is good. We could aggressively trim (which speaks to me), but one could argue that would hurt readability when written back out. We could not trim at all, but that complicates matching key and value strings in the backend.

I'd vote we accept this PR and discuss during the next meeting ... I've got a couple more PRs lined up after this one :)

Reply:
Didn't see your response a minute later ... I don't mind switching to "aggressive trimming" for this PR. But a group discussion is warranted.